src/main/capi/prepared-c.cpp

```diff
@@ -113,19 +113,23 @@ static duckdb::string duckdb_parameter_name_internal(duckdb_prepared_statement p
 	if (!wrapper || !wrapper->statement || wrapper->statement->HasError()) {
 		return duckdb::string();
 	}
-	if (index > wrapper->statement->named_param_map.size()) {
+	auto &named = wrapper->statement->named_param_map;
+	if (index == 0 || index > named.size()) {
 		return duckdb::string();
 	}
-	for (auto &item : wrapper->statement->named_param_map) {
-		auto &identifier = item.first;
-		auto &param_idx = item.second;
-		if (param_idx == index) {
-			// Found the matching parameter
-			return identifier;
+
+	auto &cache = wrapper->param_index_to_name;
+	if (cache.size() != named.size() + 1) {
```

Comment:
Can we turn this into a function and then move that cache initialization function to where we create the prepared statement wrapper? So the places where we do `auto wrapper = new PreparedStatementWrapper();` followed by some try-catch to prepare the statements.